[Ethan Wong Goon Hong] iP

EthanWong6362 opened the pull request:
> Computers are Stupid, but Duke is not.  - ~~not~~ Pablo Picasso

# Why should you use Duke
The human mind is many things:
* Creative
* Complex
* (insert another word that starts with c here)

One thing it is not, however, is organised.

## But why use your brain when you could simply choose not to
By clicking [this link](https://www.youtube.com/watch?v=Tt7bzxurJ1I) you can get your very own task manager for the low low price of **FREE**!!!

Our features include: 
- [x] Add tasks
- [x] Delete tasks
- [x] Find tasks
- [ ] Do your homework
- [ ] Make your parents proud of you even though you're always stuck in a dark corner of your room, wasting away the best times of your youth on the internet instead of doing something worth anything in your life

(we're still workshopping the last 2)

## What are you waiting for?
Get your very own Duke now!
```
    ____________________________________________________________
    Hello, I'm Duke!
    How may I be of service to you?
    ____________________________________________________________
```

Commit:
Add Gradle support

EthanWong6362 committed:
level-1

EthanWong6362 committed:
level-2

EthanWong6362 committed:
level-3

EthanWong6362 committed:
level-4

EthanWong6362 committed:
level-5

EthanWong6362 committed:
level-5

EthanWong6362 committed:
level-6

EthanWong6362 committed:
A-Enums

EthanWong6362 committed:
deleted unnecessary stuff

EthanWong6362 committed:
Adds save/read from file function

EthanWong6362 committed:
Adds date parsing functionality (Deadline)

EthanWong6362 committed:
Adds more OOP

EthanWong6362 committed:
Packages everything

EthanWong6362 committed:
Delete Test.java

EthanWong6362 committed:
Adds Find feature

EthanWong6362 committed:
Fixes code to Java standards

EthanWong6362 committed:
*ignore*

EthanWong6362 committed:
*ignore*

EthanWong6362 committed:
Merge branch 'branch-A-CodingStandard'

EthanWong6362 committed:
Tests methods in TaskList

EthanWong6362 committed:
Creates Jar file

EthanWong6362 committed:
ui testing

KishendranVendarKon commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,33 @@
+
+    public void run() {
+        ui.firstWelcome();
+        boolean cont = true;
```
Boolean should sound like a boolean e.g. toContinue

KishendranVendarKon commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,82 @@
+                    }
+                }
+                taskReader.close();
+            } else {
```
I think perhaps it may look cleaner to just omit the else block since it's empty

KishendranVendarKon commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+            return;
+        }
+
+        String line = "____________________________________________________________";
```
Perhaps you could declare line an indent as constants as they are used in more than one function?

KishendranVendarKon reviewed (commented):
Looks good. I like how clean your code is.

EthanWong6362 committed:
Adds javadoc descriptions

EthanWong6362 committed:
idk why this is here ngl

EthanWong6362 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

EthanWong6362 committed:
Adds gradle support

sivayogasubramanian commented on build.gradle:
```diff
@@ -0,0 +1,46 @@
+    toolVersion = '8.29'
+}
+
+run{
```
Forgot the space here

```suggestion
run {
```

sivayogasubramanian commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,54 @@
+public class Deadline extends Task {
+    private LocalDateTime deadline;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("MMM d yyyy, hh:mm a");
+    private static final DateTimeFormatter SAVE_FORMAT = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
```
I think it's good practice to have static objects on top.
```suggestion
    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("MMM d yyyy, hh:mm a");
    private static final DateTimeFormatter SAVE_FORMAT = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
    private LocalDateTime deadline;
```

sivayogasubramanian commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,39 @@
+        }
+    }
+
+    public static void main(String[] args) {
+        new Duke(Paths.get("data", "test.txt")).run();
+    }
```
The `main` method should be the very first thing in this class (below the constructor). Improves readability.

sivayogasubramanian commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,92 @@
+                    String[] taskComps = taskln.split("\\|");
+                    if (taskComps[0].equals("T")) {
+                        tasks.add(new Todo(taskComps[2], Boolean.parseBoolean(taskComps[1])));
+                    } else if (taskComps[0]. equals("D")) {
```
No need a space here
```suggestion
                    } else if (taskComps[0].equals("D")) {
```

EthanWong6362 committed:
Changes to adhere to checkstyle

sivayogasubramanian commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,48 @@
+    public void testAddTasks() {
+        int n = 7;
+        int init = TASKS.sizeOf();
+        for(int i = 0; i < n; i++) {
```
Add space here
```suggestion
        for (int i = 0; i < n; i++) {
```

sivayogasubramanian commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,38 @@
+ * An enum of all available commands + invalid command.
+ */
+public enum Command {
+    LIST("list"),
+    FIND("find"),
+    DONE("done"),
+    DELETE("delete"),
+    CLEAR("clear"),
+    TODO("todo"),
+    DEADLINE("deadline"),
+    EVENT("event"),
+    BYE("bye"),
+    INVALID("invalid");
```
It might be good to order this in alphabetical order.

EthanWong6362 committed:
*ignore*

EthanWong6362 committed:
*ignore*

sivayogasubramanian commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,193 @@
+        this.storage.initialiseTasks(tasks);
+    }
+
+    public int sizeOf() {
```
Might be good to give this a better name
```suggestion
    public int getTasklistSize() {
```

EthanWong6362 committed:
Uses varargs (from a long time ago)

sivayogasubramanian commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,193 @@
+     *
+     * @param index Index of task.
+     */
+    public void doneTask(int index) {
```
Might be good to give this a better name
```suggestion
    public void markTaskAsDone(int index) {
```

sivayogasubramanian reviewed (commented):
You code is mostly clean with a few coding style problems here and there. Otherwise, LGTM!

EthanWong6362 committed:
Adds JavaFX GUI

EthanWong6362 committed:
Add Assert checks

There are no assert checks for variables in Parser and Storage.

An incorrect instantiation of these variables could cause null pointer errors.

Let's add checks before these variables are used to ensure they are not null.

EthanWong6362 committed:
Improve method names

Some methods currently have ambiguous names.

This lowers the readability of the code.

Let's refactor them to better reflect the functionality of these methods.

EthanWong6362 committed:
Merge pull request #1 from EthanWong6362/branch-A-Assertions

Add Assert checks

EthanWong6362 committed:
Merge pull request #2 from EthanWong6362/branch-A-CodeQuality

Improve method names

EthanWong6362 committed:
Allow shortforms for commands

EthanWong6362 committed:
Set theme jekyll-theme-slate

EthanWong6362 committed:
update README

EthanWong6362 committed:
update build.gradle
